### PR TITLE
Deselect btns

### PR DESCRIPTION
**Pull request by SULAIMAN-100**
Before sending your Pull Request, be sure to check that you are making the pull request from your branch to the dev branch! Please do not try to push directly into branches on the main repo.

- @petemcfarlane || @CanaryWharf || @samkenny  we tried to create a function to deselect the buttons from the student side as a toggle to each button individually, but it seems to be not perfect as it should work, the problem is when we select one button we able to deselect it correctly, however, when we select the first time and try to update by clicking  another button it deselects the previous btn.
Can you help Please?

### CODE REVIEW

File: client/src/Pages/SkillsTracker.js [hunk 1]
```diff
@@ -12,11 +12,11 @@ export default function Html({ skill }) {
           throw data;
         }
         setLearningObjectives(data);
+        console.log(data);
```
Comment (cgkr) [inline client/src/Pages/SkillsTracker.js]: Perhaps this slipped through?

File: client/src/Pages/SkillsTracker.js [hunk 2]
```diff
@@ -38,28 +38,57 @@ export default function Html({ skill }) {
     setLearningObjectives(
       learningObjectives.map((obj) => {
         if (obj.id === id) {
+          console.log( {obj} );
```
Comment (cgkr) [inline client/src/Pages/SkillsTracker.js]: Perhaps this slipped through?

**Review by cgkr — APPROVED** [review]

File: server/api.js [hunk 3]
```diff
@@ -156,6 +157,33 @@ router.post("/abilities", authorization, async (req, res) => {
...
+    const learning_obj_id = Number(req.body.learning_obj_id);
+    const student_id = Number(req.body.student_id);
+    const ability = req.body.ability
+    console.log(learning_obj_id);
+    console.log(student_id);
+    console.log(ability);
+    const queryDeselect = `update achievements set  ability= $1
+                      where learning_obj_id = $2 and student_id =$3`; 
+        await Connection.query(queryDeselect, [ability, learning_obj_id, student_id]);
+        res.json("deselected")
```
Comment (petemcfarlane) [inline server/api.js:173]: This seem like a repeat of https://github.com/CodeYourFuture/knowledge-checklist/pull/13/files#diff-a813eb0ded684fcb4363a4e760c34103f7db7001843ad3706d560397212010f8R142-R143, is there a reason you can't just use that endpoint ?

File: server/api.js [hunk 4]
```diff
@@ -156,6 +157,33 @@ router.post("/abilities", authorization, async (req, res) => {
     res.json("inserted");
   }
 });
+//<-------------------deselect learning obj btns---------------------->
+
+router.put("/deselect", async (req, res)=>{
```
Comment (petemcfarlane) [inline server/api.js:162]: 🔴 notice this doesn't have any `authorisation` middleware, so anyone (don't have to be logged in) can deselect the abilities of any user

File: client/src/components/ProgressTrackingButtons.js [hunk 5]
```diff
@@ -1,23 +1,47 @@
...
-        classNameBtn={ability === 0 ? "btn-red" : ""}
-        onClick={() => updateAbility(0)}
+        classNameBtn={ability === 0 ? "btn-red" : "btn-white"}
+        onClick={() =>
+          ability == null
+            ? updateAbility(0)
+            : ability == 0
+            ? deselect(student_id, null, learningObjId)
+            : updateAbility(0)
+        }
```
Comment (petemcfarlane) [inline client/src/components/ProgressTrackingButtons.js:21]: This (and the other below) ternaries are very confusing to me. I'm not sure you need a different api endpoint and specific `deselect()` function (can't you just use `updateAbility(null)`?) but even if you do have some reason to use a new `deselect()` function, can't you do
```suggestion
        onClick={() =>
          ability === 0
            ? deselect(student_id, null, learningObjId)
            : updateAbility(0)
        }
```

**Review by petemcfarlane — CHANGES_REQUESTED** [review]
Hey sorry for the wait. I've left some feedback, see comments. Namely, do you even need the `deselect` endpoint (can't you use the existing one with a `null` score? And we should make sure the endpoint requires `authorization`. And perhaps we can make the ternaries a bit easier to read 👍

File: server/api.js (hunk 3 above)
Comment (AltomHussain) [reply server/api.js:173]: @petemcfarlane we have made suggested changes can you please check? Also we removed ```Number()``` from `const ability = req.body.ability` from the ability enpoint in order for us to get the null value from `updateAbility(null)` is this okay or is there a better way?
Thanks.

**Pull request merged**